src/http/http.cpp

```
@@ -98,7 +99,7 @@ namespace http {
 			case 504: return GATEWAY_TIMEOUT;
 			case 505: return HTTP_VERSION_NOT_SUPPORTED;
 			case 508: return LOOP_DETECTED;
-			default: return INTERNAL_SERVER_ERROR;
+			default: return _NOT_IMPLEMENTED;
```

lets throw here instead. this was a mistake before and we actually do want to throw in this case so we catch invalid request/cgi status codes